core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -101,6 +119,25 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                 if (AUTO.asString().equalsIgnoreCase(fuzziness)) {
                     return AUTO;
                 }
+                else if (fuzziness.toUpperCase(Locale.ROOT).startsWith(AUTO.asString() + ":")) {
+                    String[] fuzzinessLimit = fuzziness.substring(AUTO.asString().length() + 1).split(",");
+                    if (fuzzinessLimit.length == 2) {
+                        try {
+                            int lowerLimit = Integer.parseInt(fuzzinessLimit[0]);
+                            int highLimit = Integer.parseInt(fuzzinessLimit[1]);
+
+                            if (lowerLimit < 0 || highLimit < 0 || lowerLimit > highLimit) {
+                                throw new ElasticsearchParseException("Auto fuzziness wrongly configured");
+                            }
+                            return new Fuzziness("AUTO", lowerLimit, highLimit);
```

Comment:
`fuzziness.toUpperCase(Locale.ROOT)` it happens inside the constructor (with String) as it was already present. The same behaviour was already used in line 45 `    public static final Fuzziness AUTO = new Fuzziness("AUTO");`  . I think that, probably, it should be kept inside the constructor and not outside. What do you think? Happy to refactor it if you think we should move it outside.